src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -0,0 +1,1072 @@
+    static EntryStorage acl[kNumberOfFabrics * kEntriesPerFabric];
+
+    static EntryStorage * FindUnusedInAcl(size_t * index, const FabricIndex * fabricIndex)
+    {
+        for (auto & storage : acl)
+        {
+            if (!storage.InUse())
+            {
+                if (index != nullptr)
+                {
+                    if (fabricIndex != nullptr)
+                    {
+                        *index = 0;
+                        for (auto & storage2 : acl)
+                        {
+                            if (&storage2 == &storage)
+                            {
+                                break;
+                            }
+                            else if (storage2.mFabricIndex == storage.mFabricIndex)
+                            {
+                                (*index)++;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        *index = &storage - acl;
+                    }
+                }
+                return &storage;
+            }
+        }
+        return nullptr;
+    }
+
+    static EntryStorage * FindUsedInAcl(size_t index, const FabricIndex * fabricIndex)
+    {
+        index = FilterIndex(index, fabricIndex);
+        if (index < ArraySize(acl))
+        {
+            auto storage = acl + index;
+            if (storage->InUse())
+            {
+                return storage;
+            }
+        }
+        return nullptr;
+    }
+
+public:
+    // Pool support
+    static EntryStorage pool[kEntryStoragePoolSize];
+
+    static EntryStorage * Find(EntryStorage * candidate)
+    {
+        if (candidate && candidate->InPool())
+        {
+            return candidate;
+        }
+        for (auto & storage : pool)
+        {
+            if (!storage.InUse())
+            {
+                return &storage;
+            }
+        }
+        return nullptr;
+    }
+
+    bool InPool() const
+    {
+        constexpr auto end = pool + ArraySize(pool);
+        return pool <= this && this < end;
+    }
+
+public:
+    EntryStorage() = default;
+
+    void Init()
+    {
+        if (!mInUse)
+        {
+            Clear();
+            mInUse = true;
+        }
+    }
+
+    bool InUse() const { return mInUse; }
+
+    void Release()
+    {
+        if (InPool())
+        {
+            mInUse = false;
+        }
+    }
+
+    void Clear()
+    {
+        mInUse       = false;
+        mFabricIndex = chip::kUndefinedFabricIndex;
+        mAuthMode    = AuthMode::kPase;
+        mPrivilege   = Privilege::kView;
+        for (auto & subject : mSubjects)
+        {
+            subject.Clear();
+        }
+        for (auto & target : mTargets)
+        {
+            target.Clear();
+        }
+    }
+
+private:
+    static size_t FilterIndex(size_t index, const FabricIndex * fabricIndex)
```

Comment:
This function's semantics need to be documented.  What is it trying to do?

Reply:
This is part of fabric indexing support.  Not currently used. Still need to write tests for it, and documentation.